packages/bierzo-wallet/src/routes/transactions/components/sorting.ts

```diff
@@ -0,0 +1,45 @@
+import { ProcessedTx } from '../../../store/notifications';
+
+export interface SortingStateProps {
+  readonly onSort: (orderBy: TxsOrder, order: SortOrder) => () => void;
+  readonly orderBy: TxsOrder;
+  readonly order: SortOrder;
+}
+
+export const ORDER_ASC = 1;
+export const ORDER_DESC = -1;
+export type SortOrder = 1 | -1;
+
+export const TX_AMOUNT_COLUMN = 'Amount';
+export const TX_DATE_COLUMN = 'Date';
+export type TxsOrder = 'Date';
+
+export const filterTxsBy = (
+  txs: ReadonlyArray<ProcessedTx>,
+  rowsPerPage: number,
+  pageNumber: number,
+  orderBy: TxsOrder,
+  order: SortOrder,
+): ReadonlyArray<ProcessedTx> => {
+  const orderedTxs = txs.slice(0).sort((a: ProcessedTx | undefined, b: ProcessedTx | undefined) => {
```

Comment:
- When `txs` is of type `ReadonlyArray<ProcessedTx>`, why are the elements `a` and `b` in the compare function of type `ProcessedTx | undefined`?
- `slice(0)` is there to make a mutable copy, right?